### PR TITLE
Improvements to CreateKeyEmailFragment

### PR DESCRIPTION
**Pull request by Hiperzone**

### CODE REVIEW

**Comment by Hiperzone** [comment]
Will check tomorrow why the build failed

**Comment by dschuermann** [comment]
Will look into this pull request later, stay tuned.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java [hunk 1]
```diff
@@ -95,9 +89,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         View view = inflater.inflate(R.layout.create_key_email_fragment, container, false);
 
         mEmailEdit = (EmailEditText) view.findViewById(R.id.create_key_email);
-        mBackButton = view.findViewById(R.id.create_key_back_button);
-        mNextButton = view.findViewById(R.id.create_key_next_button);
-        mEmailsRecyclerView = (RecyclerView) view.findViewById(R.id.create_key_emails);
+        View mBackButton = view.findViewById(R.id.create_key_back_button);
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: if the variables are no longer class variables, remove the prefix ``m``

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java [hunk 2]
```diff
@@ -133,9 +127,10 @@ public void onClick(View v) {
                     addEmail();
                 }
             });
-        }
-        if (mAdditionalEmailModels.isEmpty() && mCreateKeyActivity.mAdditionalEmails != null) {
-            mEmailAdapter.addAll(mCreateKeyActivity.mAdditionalEmails);
+
+            if (mCreateKeyActivity.mAdditionalEmails != null) {
+                mEmailAdapter.addAll(mCreateKeyActivity.mAdditionalEmails);
+            }
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: Why did you remove ``mAdditionalEmailModels.isEmpty()``? I think it prevented a crash so I added it recently.
Also please fix the indent here.
Maybe you should better revert these lines to the previous version.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java [hunk 3]
```diff
@@ -143,6 +138,68 @@ public void onClick(View v) {
...
+        return !(email.length() == 0 || !emailPattern.matcher(email).matches());
+    }
+
+    /**
+     * Checks for duplicated emails inside the additional email adapter.
+     *
+     * @param email
+     * @return
+     */
+    private boolean isEmailDuplicatedInsideAdapter(String email) {
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: Isn't this already checked by the recycler list? I already get an error when adding a duplicated item in the current development branch.

**Comment by dschuermann** [comment]
Regarding email validation:
I personally don't like validating email addresses with regexp (also https://news.ycombinator.com/item?id=5763327). We already have the small indicator on the right that indicates if something is wrong (okay looks like it's broken right now: https://github.com/open-keychain/open-keychain/issues/1231 :/ )

**Comment by Hiperzone** [comment]
As we discussed earlier there was a case where duplicated emails were allowed so I decided to refactor the whole code.
The code you added was removed because when I changed the code, everything seemed to work fine for all the test scenarios that I could remember (Tested on Kitkat and lollipop devices). The tests included screen rotation and activity recreation with and without email data.

When I coded the new changes I did not have the changes that you made later and since everything was working fine I did not take it into consideration.

With the fixes I applied to createView, the bug you encountered may have been fixed, but instead of doing assumptions, I'm going to do new tests today and also include a few stability fixes to the CloudImportService :)

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java (hunk 1 above)
Comment (Hiperzone) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: Will be done today. my bad :)

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java (hunk 2 above)
Comment (Hiperzone) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java]: Will do new tests today for the condition but it may be not needed since i changed the code to take null cases into consideration when adding emails from the activity to the adapter.

File: OpenKeychain/src/main/res/layout/create_key_email_fragment.xml [hunk 4]
```diff
@@ -32,6 +32,7 @@
                 android:layout_marginTop="16dp"
                 android:layout_marginBottom="8dp"
                 android:imeOptions="actionNext"
+                android:inputType="textEmailAddress"
```
Comment (dschuermann) [inline OpenKeychain/src/main/res/layout/create_key_email_fragment.xml]: not needed due to https://github.com/Hiperzone/open-keychain/blob/development/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/widget/EmailEditText.java#L53

File: OpenKeychain/src/main/res/layout/add_email_dialog.xml [hunk 5]
```diff
@@ -21,6 +21,7 @@
         android:layout_marginTop="16dp"
         android:hint="@string/label_email"
         android:imeOptions="actionNext"
+        android:inputType="textEmailAddress"
```
Comment (dschuermann) [inline OpenKeychain/src/main/res/layout/add_email_dialog.xml]: not needed due to https://github.com/Hiperzone/open-keychain/blob/development/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/widget/EmailEditText.java#L53

**Comment by Hiperzone** [comment]
Updated, if you need me to do more changes, don't hesitate to tell me :)
